src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/ApplicationsTransformer.kt

```diff
@@ -116,4 +125,24 @@ class ApplicationsTransformer(
       else -> ApplicationStatus.inProgress
     }
   }
+
+  private fun getStatusFromSummary(entity: DomainApplicationSummary): ApplicationStatus {
+    if (entity is DomainApprovedPremisesApplicationSummary) {
+      return when {
+        entity.getSubmittedAt() != null && entity.getLatestAssessmentDecision() == AssessmentDecision.REJECTED -> ApplicationStatus.rejected
+        entity.getSubmittedAt() != null && entity.getLatestAssessmentDecision() == AssessmentDecision.ACCEPTED && !entity.getHasPlacementRequest() -> ApplicationStatus.pending
+        entity.getSubmittedAt() != null && entity.getLatestAssessmentDecision() == AssessmentDecision.ACCEPTED && !entity.getHasBooking() -> ApplicationStatus.awaitingPlacement
+        entity.getSubmittedAt() != null && entity.getLatestAssessmentDecision() == AssessmentDecision.ACCEPTED && entity.getHasBooking() -> ApplicationStatus.placed
+        entity.getLatestAssessmentHasClarificationNotesWithoutResponse() -> ApplicationStatus.requestedFurtherInformation
+        entity.getSubmittedAt() !== null -> ApplicationStatus.submitted
+        else -> ApplicationStatus.inProgress
```

Comment:
Probs not for now, but I'm wondering if all this stuff here has a perfomance impact? Could we manage statuses via a state machine or similar and record them directly in the database?

Reply:
Having a `status` column on the applications table would definitely make reading simpler and more efficient.  It is of course a trade off as if we want to change the status logic (as we did recently to add more statuses) at some point we have to figure out a migration process too.